app/src/main/kotlin/at/bitfire/davdroid/sync/worker/SyncWorkerManager.kt

```diff
@@ -285,4 +291,28 @@ class SyncWorkerManager @Inject constructor(
             }
     }
 
+    /**
+     * Listener that enqueues a push registration worker when the collection list changes.
+     */
+    class CollectionsListener @Inject constructor(
+        private val workerManager: SyncWorkerManager
+    ): DavCollectionRepository.OnChangeListener {
+
+        override fun onCollectionsChanged(account: Account?) {
+            account?.let { workerManager.enqueueOneTimeAllAuthorities(it) }
+        }
+
+    }
+
+    /**
+     * Hilt module that registers [CollectionsListener] in [DavCollectionRepository].
+     */
+    @Module
+    @InstallIn(SingletonComponent::class)
+    interface SyncWorkerManagerModule {
+        @Binds
+        @IntoSet
+        fun listener(impl: CollectionsListener): DavCollectionRepository.OnChangeListener
+    }
```

Comment:
I think it should be in a separate file, because `SyncWorkerManager`'s task is only to manage (enqueue, cancel) syncs.

The functionality that a method of `SyncWorkerManager` shall be called when collections change is a separate function and should be in a separate file.